SignalServiceKit/src/Storage/OWSStorage.h

```diff
@@ -6,24 +6,34 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+extern NSString *const StorageIsReadyNotification;
+
 @class YapDatabaseExtension;
 
 @interface OWSStorage : NSObject
 
 - (instancetype)init NS_UNAVAILABLE;
 - (instancetype)initStorage NS_DESIGNATED_INITIALIZER;
 
-- (void)setDatabaseInitialized;
+// Returns YES if _ALL_ storage classes have completed both their
+// sync _AND_ async view registrations.
++ (BOOL)isStorageReady;
+
+/**
```

Comment:
`OWSStorage` will have a bunch of class methods for:

* ...coordinating and observing storage setup/readiness.
* ...resetting storage.